### PR TITLE
ignore case in original extension

### PR DESCRIPTION
**Pull request by daichengxin**
### **PR Type**
enhancement, documentation

___

### **Description**
- Enhanced the `_removesuffix` function to handle suffixes in a case-insensitive manner.
- Updated the logic for file extension conversion to be case-insensitive.
- Revised the help text in `parse_sdrf.py` to reflect the case-insensitive nature of extension conversion.
- Bumped the version from 0.0.31 to 0.0.32.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>__init__.py</strong><dd><code>Bump version to 0.0.32</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

sdrf_pipelines/__init__.py

- Updated the version from 0.0.31 to 0.0.32.

</details>

  </td>
  <td><a href="https://github.com/bigbio/sdrf-pipelines/pull/183/files#diff-2597b8d13a430b6f70d1f7403f67394b82fcd3249affce610251564dbd939806">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>openms.py</strong><dd><code>Implement case-insensitive extension handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

sdrf_pipelines/openms/openms.py

<li>Modified <code>_removesuffix</code> function to ignore case when checking suffix.<br> <li> Updated logic to handle case-insensitive extension conversion.<br>

</details>

  </td>
  <td><a href="https://github.com/bigbio/sdrf-pipelines/pull/183/files#diff-8229e69a289dbe40b85ae9ca60ee48b888d2efe1dee568f24cd40d6f49b3319d">+4/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>parse_sdrf.py</strong><dd><code>Update documentation for case-insensitive extension conversion</code></dd></summary>
<hr>

sdrf_pipelines/parse_sdrf.py

<li>Updated help text to indicate case-insensitive extension conversion.<br>

</details>

  </td>
  <td><a href="https://github.com/bigbio/sdrf-pipelines/pull/183/files#diff-927b09551fffe1ba8cb5f47e8e2ee4230f7ab50cbcfa917310ac358c0a1e35fd">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

**Review by ypriverol — APPROVED** [review]

File: sdrf_pipelines/openms/openms.py [hunk 1]
```diff
@@ -59,7 +60,7 @@ def _removesuffix(x: str, suffix: str, /) -> str:
 
     raw_bkp = raw
     for current_extension, target_extension in extension_convert_dict.items():
-        if raw.endswith(current_extension):
+        if raw.lower().endswith(current_extension.lower()):
             raw = _removesuffix(raw, current_extension)
             raw += target_extension
             if not any(raw.endswith(x) for x in possible_extension):
```
Comment (jpfeuffer) [inline sdrf_pipelines/openms/openms.py:66]: I think you will need to change this,too in case the user uses RAW and does no conversion of it.

**Review by jpfeuffer — CHANGES_REQUESTED** [review]

**Comment by jpfeuffer** [comment]
@ypriverol I think this was too quick

File: sdrf_pipelines/openms/openms.py (hunk 1 above)
Comment (daichengxin) [reply sdrf_pipelines/openms/openms.py:66]: Here I assume that the final openms experimental design file has the same suffix. Is that right? For example
Example 1: `raw:mzML`: 
1.raw    -> 1.mzML
2.Raw    -> 2.mzML
3.RAW   -> 3.mzML

Example 2: `raw:RAW`:
1.raw -> 1.RAW
2.Raw -> 2.RAW
3. RAW -> 3.RAW

Comment (daichengxin) [reply sdrf_pipelines/openms/openms.py:66]: And also we assumed only suppoted four format: `"raw", "mzML", "mzml", "d"` https://github.com/bigbio/sdrf-pipelines/blob/d87d22c7b763f290d4cebb7c336a66af205be527/sdrf_pipelines/openms/openms.py#L66-L67

Comment (jpfeuffer) [reply sdrf_pipelines/openms/openms.py:66]: If I am not mistaken your example raw:RAW will fail.
If you only allow ".raw" as possible extension you have to add raw:raw as a default conversion to the pipeline which is super weird and not expected.

Comment (jpfeuffer) [reply sdrf_pipelines/openms/openms.py:66]: Why not allow the possible extensions case insensitive then?
Would solve a lot of problems.

Comment (jpfeuffer) [reply sdrf_pipelines/openms/openms.py:66]: It depends a bit on if downstream tools allow different cases. (e.g. does DIANN allow .D?)

Comment (daichengxin) [reply sdrf_pipelines/openms/openms.py:66]: Good ideas. I need to test it. Wait a while. Also is ProteomicsLFQ allowed different cases? For example input` .mzML` file, but in the lab file it's `mzml`

Comment (jpfeuffer) [reply sdrf_pipelines/openms/openms.py:66]: ProteomicLFQ expects *exact* matches of the full filename of the files that you pass as -in and what is written in the experimental design (including extension, excluding paths).

Comment (daichengxin) [reply sdrf_pipelines/openms/openms.py:66]: DIA-NN don't support `.D` format. So I suggest the possible extensions is case-sensitive. What do you think?

Comment (daichengxin) [reply sdrf_pipelines/openms/openms.py:66]: > Is this conversion to openms experimental design even used in the DIA branch of the pipeline?

Yes. It also used in DIA branch https://github.com/bigbio/quantms/blob/b7c9b6a416723c4210effd3889dadce3d9032938/workflows/quantms.nf#L55

**Comment by daichengxin** [comment]
Another related question: Do we need to support this format ? https://github.com/bigbio/quantms/issues/430

**Comment by ypriverol** [comment]
@daichengxin I have asked how the `.dia` is obtained. Let's see what they say.

**Pull request merged**